test/TestProjects/RequestContextAllOptional-LowLevel/Generated/RequestContextAllOptionalClient.cs

```diff
@@ -52,19 +52,19 @@ public RequestContextAllOptionalClient(AzureKeyCredential credential, Uri endpoi
 
         /// <summary> No RequestBody and ResponseBody. </summary>
         /// <param name="id"> Query parameter Id. </param>
-        /// <param name="top"> Query parameter top. </param>
         /// <param name="skip"> Query parameter skip. </param>
+        /// <param name="top"> Query parameter top. </param>
         /// <param name="status"> Query parameter status. </param>
         /// <param name="context"> The request context, which can override default behaviors on the request on a per-call basis. </param>
 #pragma warning disable AZC0002
-        public virtual async Task<Response> NoRequestBodyResponseBodyAsync(int id, int? top = null, int skip = 12, string status = "start", RequestContext context = null)
+        public virtual async Task<Response> NoRequestBodyResponseBodyAsync(int id, int skip = 12, int? top = null, string status = "start", RequestContext context = null)
```

Comment:
Why do these two change positions?  Aren't they both optional?

Reply:
[Skip](https://github.com/Azure/autorest.csharp/blob/feature/v3/test/TestProjects/RequestContextAllOptional-LowLevel/RequestContextAllOptional.json#L38) is required while [top](https://github.com/Azure/autorest.csharp/blob/feature/v3/test/TestProjects/RequestContextAllOptional-LowLevel/RequestContextAllOptional.json#L23) is not.

It is a good point however that both have default values. 

We can either strengthen the order here, e.g.:
1. Required without default values
2. Required with default values
3. Optional with default values

or use default value instead of `required` as criteria.

Reply:
Yes, from the point of view of a developer using our clients, if we provide a default value for something, it's optional to them.  Let's use whether or not it has a default value to indicate whether it's "required" or "optional."

Reply:
> Skip is required while top is not.

This sounds like it's not a standard swagger then - per @tg-msft, it sounds like in real life, top (i.e. specifying the page size) might be required and skip not.  Given that, I'm ok with this change.